sympy/physics/control/tests/test_lti.py

```diff
@@ -0,0 +1,233 @@
+        Matrix([7 / (p**9 + 9*p)])
+    )
+    H2 = TransferFunctionMatrix(
+        Matrix([5 / (s + 2)])
+    )
+    H = TransferFunctionMatrix(
+        Matrix([35 / ((s + 2) * (p**9 + 9*p))])
+    )
+    assert H1.series(H2).G == H.G
+
+
+def test_TransferFunctionMatrix_parallel():
+    tfm1 = TransferFunctionMatrix(
+        Matrix([1 / (a * s)])
+    )
+    tfm2 = TransferFunctionMatrix(
+        Matrix([(b + s) / (a + s)])
+    )
+    expect = TransferFunctionMatrix(
+        Matrix([(a + s + a * s * (b + s)) / (a * s * (a + s))])
+    )
+    assert together(tfm1.parallel(tfm2).G) == together(expect.G)
+
+    H1 = TransferFunctionMatrix(
+        Matrix([7 / (p**9 + 9*p)])
+    )
+    H2 = TransferFunctionMatrix(
+        Matrix([5 / (s + 2)])
+    )
+    H = TransferFunctionMatrix(
+        Matrix([7 / (p**9 + 9*p) + 5 / (s + 2)])
+    )
+    assert together(H1.parallel(H2).G) == together(H.G)
+
+
+def test_TransferFunctionMatrix_negate():
+    G1 = Matrix([s / (s + 1)])
+    tf1 = TransferFunctionMatrix(G1)
+    assert tf1.neg() == TransferFunctionMatrix(
+        Matrix([-s / (s + 1)])
+    )
+
+    G2 = Matrix([[(4*p - 10) / (2*p + 1), 3 / (p + 2)],
+            [1 / (2*p**2 + 5*p + 2), (p + 1) / (p**2 + 4*p + 4)]])
+    tf2 = TransferFunctionMatrix(G2)
+    assert tf2.neg() == TransferFunctionMatrix(
+        Matrix([[(10 - 4*p) / (2*p + 1),  -3 / (p + 2)],
+            [-1 / (2*p**2 + 5*p + 2), (-p - 1) / (p**2 + 4*p + 4)]])
+    )
+
+
+def test_TransferFunctionMatrix_is_proper():
+    G1 = Matrix([2*p / (2 + p**2 - p), p / (1 + p**3)])
+    assert TransferFunctionMatrix(G1).is_proper
+
+    G2 = Matrix([2*s**6/(2 + s**2 - s), (s**4)/(1 + s**2)])
+    assert not TransferFunctionMatrix(G2).is_proper
+
+
+def test_TransferFunctionMatrix_is_strictly_proper():
+    G1 = Matrix([2*s**6 / (2 + s**2 - s), (s**4) / (1 + s**2)])
+    assert not TransferFunctionMatrix(G1).is_strictly_proper
+
+    G2 = Matrix([[(4*p - 10) / (2*p**3 + 1), 3 / (p + 2)],
+            [1 / (2*p**2 + 5*p + 2), (p + 1) / (p**2 + 4*p + 4)]])
+    assert TransferFunctionMatrix(G2).is_strictly_proper
+
+
+def test_TransferFunction_construction():
+    tf = TransferFunction(s + 1, s**2 + s + 1)
+    assert tf.num == (s + 1)
+    assert tf.den == (s**2 + s + 1)
+    assert tf.args == (s + 1, s**2 + s + 1)
+
+    tf1 = TransferFunction(s + 4, s - 5)
+    assert tf1.num == (s + 4)
+    assert tf1.den == (s - 5)
+    assert tf1.args == (s + 4, s - 5)
+
+    # using different polynomial variables.
+    tf2 = TransferFunction(p + 3, p**2 - 9)
+    assert tf2.num == (p + 3)
+    assert tf2.den == (p**2 - 9)
+    assert tf2.args == (p + 3, p**2 - 9)
+
+    tf3 = TransferFunction(p**3 + 5*p**2 + 4, p**4 + 3*p + 1)
+    assert tf3.args == (p**3 + 5*p**2 + 4, p**4 + 3*p + 1)
+
+    # no pole-zero cancellation on its own.
+    tf4 = TransferFunction((s + 3)*(s - 1), (s - 1)*(s + 5))
+    assert tf4.den == (s**2 + 4*s - 5)
+    assert tf4.args == (s**2 + 2*s - 3, s**2 + 4*s - 5)
```

Comment:
SymPy does not expand the expressions by default. Why do you want to expand the provided expressions?

```
In [1]: import sympy                                                                                                                        

In [2]: s = sympy.symbols('s')                                                                                                              

In [3]: (s -1 )*(s-2)                                                                                                                       
Out[3]: (s - 2)*(s - 1)
```

Reply:
> Why do you want to expand the provided expressions?

Even I was initially confused but I decided to follow python-control here too! :)
``` python
>>> tf = ((s - 1)*(s - 2))/((s - 3)*(s - 4))
>>> tf
s^2 - 3 s + 2
--------------
s^2 - 7 s + 12
```

Reply:
This is one place not to follow a numerical library. This is the difference in how symbolics work.

Reply:
Okay, I'll keep them as they are. (no usage of `expand` here)